File: ddd-application/src/main/java/victor/training/ddd/agile/dto/ProductDto.java
package victor.training.ddd.agile.dto;

import lombok.Data;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@Data
public class ProductDto {
   private Long id;
   @NotNull
   @NotBlank
   private String code;
   @NotNull
   @NotBlank
   private String name;
   private String mailingList;
   private String poEmail;
   private String poName;
   private String poPhone;

}

File: ddd-application/src/main/java/victor/training/ddd/agile/dto/ProductDto.kt
package victor.training.ddd.agile.dto

class ProductDto {
    var id: Long? = null
    var code: String
    var name: String
    var mailingList: String
    var poEmail: String? = null
    var poName: String? = null
    var poPhone: String? = null

    constructor(id: Long?, code: String, name: String, mailingList: String) {
        this.id = id
        this.code = code
        this.name = name
        this.mailingList = mailingList
    }


    constructor(code: String, name: String, mailingList: String) {
        this.code = code
        this.name = name
        this.mailingList = mailingList
    }
}